LoopKitUI/Views/FractionalQuantityPicker.swift

```diff
@@ -56,18 +59,29 @@ struct FractionalQuantityPicker: View {
 
         self._whole = Binding(
             get: { doubleValue.wrappedValue.whole },
-            set: { newValue in
-                if newValue == guardrail.absoluteBounds.upperBound.doubleValue(for: unit) {
-                    doubleValue.wrappedValue = newValue
+            set: { newWholeValue in
+                // If the new whole value supports the same fractional value, keep it; otherwise, truncate.
+                let supportedFractionValues = fractionalValuesByWhole[newWholeValue]!
```

Comment:
Sure would love to see a unit test for this.

Reply:
Can do in a follow-up, thanks!